python/mxnet/gluon/model_zoo/text/lm.py

```diff
@@ -15,110 +15,270 @@
+
+    Embedding size is 400, and hidden layer size is 1150.
+
+    Parameters
+    ----------
+    vocab : gluon.text.Vocabulary, default None
+        Vocabulary object to be used with the language model.
+        Required when not loading from pretrained models.
+    pretrained : str or None, default None
+        The dataset name on which the pretrained model is trained. Options are 'wikitext2'.
+        If None, then no pretrained weights are loaded.
+    ctx : Context, default CPU
+        The context in which to load the pretrained weights.
+    root : str, default '~/.mxnet/models'
+        Location for keeping the model parameters.
+
+    Returns
+    -------
+    Block, gluon.text.Vocabulary
+    """
+    predefined_args = {'model_name': 'awd_lstm_lm_1150',
+                       'mode': 'lstm',
+                       'num_layers': 3,
+                       'tie_weights': True,
+                       'dropout': 0.4,
+                       'weight_drop': 0.5,
+                       'drop_h': 0.3,
+                       'drop_i': 0.65}
+    assert all(k not in kwargs for k in predefined_args), \
+           "Cannot override predefined model settings."
+    kwargs.update(predefined_args)
+    return _get_awd_rnn(400, 1150, **kwargs)
+
+
+def simple_lstm_lm_650(**kwargs):
+    r"""Simple 2-layer LSTM language model with tied embedding and output weights.
+
+    Both embedding and hidden dimensions are 650.
+
+    Parameters
+    ----------
+    vocab : gluon.text.Vocabulary, default None
+        Vocabulary object to be used with the language model.
+        Required when not loading from pretrained models.
+    pretrained : str or None, default None
+        The dataset name on which the pretrained model is trained. Options are 'wikitext2'.
+        If None, then no pretrained weights are loaded.
+    ctx : Context, default CPU
+        The context in which to load the pretrained weights.
+    root : str, default '~/.mxnet/models'
+        Location for keeping the model parameters.
+
+    Returns
+    -------
+    Block, gluon.text.Vocabulary
+    """
+    predefined_args = {'model_name': 'simple_lstm_lm_650',
+                       'mode': 'lstm',
+                       'num_layers': 2,
+                       'tie_weights': True,
+                       'dropout': 0.5}
+    assert all(k not in kwargs for k in predefined_args), \
+           "Cannot override predefined model settings."
+    kwargs.update(predefined_args)
+    return _get_simple_rnn(650, 650, **kwargs)
+
+
+def simple_lstm_lm_1500(**kwargs):
+    r"""Simple 2-layer LSTM language model with tied embedding and output weights.
+
+    Both embedding and hidden dimensions are 1500.
+
+    Parameters
+    ----------
+    vocab : gluon.text.Vocabulary, default None
+        Vocabulary object to be used with the language model.
+        Required when not loading from pretrained models.
```

Comment:
Would different languages share the same LM?

Reply:
I see. This determines the structure of the model.

Reply:
Yes, it determines the input size of embedding layer.